internal/xds/translator/translator.go

```diff
@@ -788,10 +782,12 @@ func addXdsSecret(tCtx *types.ResourceVersionTable, secret *tlsv3.Secret) error
 	return nil
 }
 
+// addXdsCluster adds a xds cluster with args.
+// If the cluster already exists, it skips adding the cluster and returns nil
 func addXdsCluster(tCtx *types.ResourceVersionTable, args *xdsClusterArgs) error {
 	// Return early if cluster with the same name exists
 	if c := findXdsCluster(tCtx, args.name); c != nil {
-		return ErrXdsClusterExists
+		return nil
```

Comment:
this is not great from an API perspective for `addXdsCluster`, because the the old cluster with the same name may not have the same `xdsClusterArgs` as the current request, so the caller should decide how to handle this case, not this method imo

Reply:
Not a strong opinion, but should the caller call `findXdsCluster` first and decide whether they should handle this situation? 

The current approach has an implict assumption: every callers should check whether the return error isErrXdsClusterExists or not, and ignore ErrXdsClusterExists  - this cause duplications for every caller and can cause issues if the caller forget to handle this special case, like this one.

Reply:
You're absolutely right, the method is doing two many things, fine to split it up, but developers will need to make sure they call both functions when writing new logic

Reply:
For all the current callers, they just need to call `addXdsCluster` as they just simply ignore `ErrXdsClusterExists` and does nothing.

I guess this pattern comes from the Kub Client API - it totally makes sense for Kubernetes API server as the API server has concurrent writes. EG doesn't need to use the same pattern here as it's a local cache and has no concurrent writes.